src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -1295,11 +1295,19 @@ impl Step for CrtBeginEnd {
     }
 
     fn make_run(run: RunConfig<'_>) {
-        run.builder.ensure(CrtBeginEnd { target: run.target });
+        if run.target.needs_crt_begin_end() {
```

Comment:
Can you try to modify `default_condition` inside `should_run` instead?

Reply:
I don't think there's "target" information in the `should_run` context (there's no `self` argument and `ShouldRun` has no target info) to do a similar check as the one I did in `make_run`. but trying to append `default_condition(false)` (same thing with `true`) to the `path` method call in `should_run` still tries to build `crt*.o` to the host when I run `./x build src/llvm-project` which is what I'm trying to avoid/fix.

do you have some other suggestion? I'm not familiar with the `Step` API

Reply:
Aha, I forgot that this is a target dependent check.